2007-12-07  Ulrich Drepper  <drepper@redhat.com>

	[BZ #5454]
	* inet/ether_line.c: Strip hostname of whitespaces.

Index: inet/ether_line.c
===================================================================
RCS file: /cvs/glibc/libc/inet/ether_line.c,v
retrieving revision 1.6
retrieving revision 1.7
diff -u -d -p -r1.6 -r1.7
--- inet/ether_line.c	24 May 2002 08:20:58 -0000	1.6
+++ inet/ether_line.c	7 Dec 2007 16:34:36 -0000	1.7
@@ -61,19 +61,20 @@ ether_line (const char *line, struct eth
 	++line;
     }
 
-  /* Remove trailing white space.  */
-  cp = __strchrnul (line, '#');
-  while (cp > line && isspace (cp[-1]))
-    --cp;
+  /* Skip initial whitespace.  */
+  while (isspace (*line))
+    ++line;
 
-  if (cp == line)
+  if (*line == '#' || *line == '\0')
     /* No hostname.  */
     return -1;
 
+  /* The hostname is up to the next non-space character.  */
   /* XXX This can cause trouble because the hostname might be too long
      but we have no possibility to check it here.  */
-  memcpy (hostname, line, cp - line);
-  hostname [cp - line] = '\0';
+  while (*line != '\0' && *line != '#' && !isspace (*line))
+    *hostname++ = *line++;
+  *hostname = '\0';
 
   return 0;
 }
